src/sdam/state/mod.rs

```diff
@@ -254,33 +270,36 @@ impl Topology {
             // in 4.2+, we only clear connection pool if we've received a
             // "node is shutting down" error. Otherwise, we always clear the pool.
             if wire_version < 8 || error.is_shutting_down() {
-                server.clear_connection_pool();
+                server.pool.clear();
             }
+
+            self.common.message_manager.request_topology_check();
         }
     }
 
     /// Marks a server in the cluster as unknown due to the given `error`.
-    async fn mark_server_as_unknown(&self, error: Error, address: StreamAddress) {
-        let description = ServerDescription::new(address, Some(Err(error)));
-        self.update(description).await;
+    async fn mark_server_as_unknown(
+        &self,
+        error: Error,
+        server: &Server,
+        state_lock: RwLockWriteGuard<'_, TopologyState>,
+    ) -> bool {
+        let description = ServerDescription::new(server.address.clone(), Some(Err(error)));
+        self.update_and_notify(description, state_lock).await
     }
 
-    /// Updates the topology using the given `ServerDescription`. Monitors for new servers will
```

Comment:
Is there a reason for deleting the doc comment here? If it no longer applies, I think a new one should be written to replace it

Reply:
So the git diff is a little weird here. This method and doc comment were actually just moved down the file. That being said, this method does need a docstring so I added one.

This also made me realize `update_and_notify` should be handling marking the pool as ready, since it is the common entry point to updating the topology. the `pub(crate) update` is now just the internal API for doing so (as used by monitors).

Reply:
Ah yeah, sorry about that. Adding a new comment is fine.

Reply:
All these methods have docstrings now.